docs/adr/0010-transition-fxa-from-backbone-to-react.md

```diff
@@ -0,0 +1,151 @@
+  - Whatever new library or framework we choose will require a learning curve for at least part of the team. It may slow the ability to make front-end changes at first.
+
+### Option B - Move to client-side React
+
+- Description:
+  - Use client-side React.
+- Pros
+  - React is backed by a well-known company (Facebook) with a lot of dedicated resources working on the library and it's popular, has a lot of community support, and has ample documentation.
+  - The `fxa-payments-server` is already using React. We can begin thinking about reusing components across the FxA monorepo.
+  - React promotes simplicity, clarity, and predictability. We know that data exists at the top and it flows down to children components, and we know what the data is with prop and type checking. The component then takes the data and subjects it to lifecycle methods that execute in a predictable manner every time. Backbone, on the other hand, creates a JSON object that is exposed to mustache files, but we don't always know what this data is or what its types are. Backbone also oftentimes influences views via updates through a model, multiple models, and/or controllers. It's not as clear when a view or its children will be rerendered or a DOM element updated.
+  - React is fast. It bundles state updates together and figures out the fewest operations it can get away with, giving us optimization wins out of the box.
+  - React promotes declarative programming over imperative. In Backbone, we interact with the DOM directly via jQuery, _telling_ the browser exactly what to do (imperative) but with React, we _describe_ the UI with components and let React and the browser figure out how to implement it (declarative). This can be much less tedious for developers and allows React to handle state changes optimally.
+  - React promotes the idea of composition over inheritance. In Backbone, we extend views for reusability, which can cause difficulty in seeing or realizing what events and other methods or data you are inheriting (the banana, monkey, jungle problem) and overwriting. The intention of components in React is more clear and data is more deliberately passed around.
+  - React offers many integration options.
+  - React has already been audited by our security team.
+- Cons
+  - React changes fairly often. The definition of "knowing React" will continue to grow as new concepts and best practices, like React Hooks or depreciated lifecycle methods, are created or recommended.
+
```

Comment:
Is React [significantly] worse regarding dependencies?  I thought I saw that in one of our analyses

Reply:
IMO, React *tooling* involves a lot of dependencies, but that's largely code we don't deploy. (i.e. devDependencies vs dependencies)

Reply:
I'll look into this further, but my initial thoughts mirror Les's.

Reply:
If we check out [React on NPM](https://www.npmjs.com/package/react) and [Backbone on NPM](https://www.npmjs.com/package/backbone), there's 3 dependencies vs 1. [`create-react-app`](https://www.npmjs.com/package/create-react-app) lists 11, but that appears to be because of development tools. There's an interesting conversation in [this thread](https://news.ycombinator.com/item?id=19195148) about it. Of course dependencies have sub-dependencies, but I wouldn't say dependencies in production are significantly worse by any means.